donkeycar/parts/gps.py

```diff
@@ -204,30 +230,36 @@ def getGpsPosition(line, debug=False):
         if debug:
             try:
                 msg = pynmea2.parse(line)
-                logger.info(f"nmea.longitude({msg.longitude}, nmea.latitude({msg.latitude})")
             except pynmea2.ParseError as e:
-                logger.info('Ignoring NMEA parse error: {}'.format(e))
+                logger.error('NMEA parse error detected: {}'.format(e))
+                return None
 
         # Reading the GPS fix data is an alternative approach that also works
         if nmea_parts[2] == 'V':
             # V = Warning, most likely, there are no satellites in view...
-            logger.info("GPS receiver warning; position not valid")
+            logger.info("GPS receiver warning; position not valid. Ignore invalid position.")
         else:
             #
             # Convert the textual nmea position into degrees
             #
             longitude = nmea_to_degrees(nmea_parts[5], nmea_parts[6])
             latitude = nmea_to_degrees(nmea_parts[3], nmea_parts[4])
-            # print(f"Your position: lon = ({longitude}), lat = ({latitude})")
-            
+
+            if debug:
```

Comment:
I would suggest to user `logger.debug(...)` here.

Reply:
I don't want to turn on debugger for the whole application.  This is super noisy, so it needs it's own flag.

Reply:
Ok.